### DISCUSSION TITLE
Add new VCPKG_PLATFORM_TOOLSET environment variable

### DISCUSSION THREAD

**Discussion by alcroito** [microsoft/vcpkg#19246] [Ideas]
**Is your feature request related to a problem? Please describe.**
Triplet files can specify the VCPKG_VISUAL_STUDIO_PATH and VCPKG_PLATFORM_TOOLSET variables.

On the other hand, there is an environment variable for VCPKG_VISUAL_STUDIO_PATH, but none for **VCPKG_PLATFORM_TOOLSET**.

**Proposed solution**
Allow setting the VCPKG_PLATFORM_TOOLSET environment variable, which will be used by vcpkg to decide which toolset to use when building packages.

**Describe alternatives you've considered**

**Additional context**

**Comment by Neumann-A** [comment]
have you tried PR #4609 yet?

**Comment by alcroito** [comment]
I'll give it a spin tomorrow.

**Comment by alcroito** [comment]
@Neumann-A From a cursory view of the diff, it doesn't seem like the VCPKG_PLATFORM_TOOLSET environment variable is accessed anywhere, but only the variable inside the triplet file?

Also, your PR should probably be rebased on top of master.

Nevertheless I checked it out, but even installing zlib failed with the following:
https://paste.ubuntu.com/p/5q6yc7SC8X/

**Comment by Neumann-A** [comment]
hmm I fixed the merge conflict with master. Strange for me ZLIB is successful. Did you rerun bootstrap?

Could you give me the log with --debug

**Comment by Neumann-A** [comment]
ah maybe i misunderstood your request. You probably just want to switch between `v140`, `v141` and `v142`. I was more thinking about the `llvm` toolset or the `"intel"` toolset

**Comment by alcroito** [comment]
Correct, i merely want to switch between v140, v141, v142 via an environment variable.
I'll post a comment on the PR regarding the zlib thing.

**Comment by JackBoosY** [comment]
Same with https://github.com/microsoft/vcpkg/issues/11317?
